src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -795,11 +796,28 @@ private Timeout auctionTimeout(Timeout timeout, boolean shouldCacheBids) {
 
             final List<BidderError> errors = bidderResponse.getSeatBid().getErrors();
             if (CollectionUtils.isNotEmpty(errors)) {
-                for (final BidderError error : errors) {
-                    adapterMetrics.incCounter(error.getType() == BidderError.Type.timeout
-                            ? MetricName.timeout_requests
-                            : MetricName.error_requests);
-                }
+                errors.stream()
+                        .map(BidderError::getType)
+                        .distinct()
+                        .map(errorType -> {
+                            final MetricName errorMetric;
+                            switch (errorType) {
+                                case bad_input:
+                                    errorMetric = MetricName.badinput;
+                                    break;
+                                case bad_server_response:
+                                    errorMetric = MetricName.badserverresponse;
+                                    break;
+                                case timeout:
+                                    errorMetric = MetricName.timeout;
+                                    break;
+                                case generic:
+                                default:
+                                    errorMetric = MetricName.unknown_error;
+                            }
+                            return errorMetric;
+                        })
```

Comment:
Maybe it is better to move this logic to method, like it was done in AuctionHandler updateAdapterErrorMetrics()